rudaux/snapshot.py

```diff
@@ -138,14 +138,20 @@ def _get_snap_name(course_name, assignment, override):
 def get_all_snapshots(config, course_id, assignments):
     snaps = []
     for asgn in assignments:
-        snaps.append( {'due_at' : asgn['due_at'],
-                       'name' : _get_snap_name(config.course_names[course_id], asgn, None),
-                       'student_id' : None})
+        if asgn['due_at'] is not None:
```

Comment:
this change might be a bit tricky. Consider an assignment with no overrides but the instructor forgot to set the deadline -- we don't want to silently ignore that problem.

But maybe that would have thrown an exception elsewhere anyway?

Reply:
Might be worth raising here if there is an assignment that doesn't have any overrides and doesn't have due_at/etc. Then from here on out we can assume if due_at is missing it's because their are overrides

Reply:
I made it so it throws an error here if due_at is missing and there are no overrides in commit 465bc4dd4035cb95e1dc15adac571a6a5d154ca9.